gen3/bin/iam-serviceaccount.sh

```diff
@@ -0,0 +1,604 @@
+#!/bin/bash
+# 
+# After EKS 1.13, AWS allows OIDC identity provider to interact with serviceaccount in k8s
+# this would be helpful when manipulating permissions for pod directly.
+# With this implementation, there should not be the need to creeate users and 
+# key secret pair to access things in AWS, but insted roles will be applied to 
+# pods directly through service accounts.
+# Additionally, we could potentially remove certain roles attached to k8s workers.
+#
+# This script is intended to automate certain steps for this implementation, and ease 
+# the burden of creating policy and roles directly on the console or through awscli
+#
+# For more information visit the folowing link
+# https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/
+
+source "${GEN3_HOME}/gen3/lib/utils.sh"
+gen3_load "gen3/lib/kube-setup-init"
+
+SCRIPT=$(basename ${BASH_SOURCE[0]})
+ACOUNT_ID=$(aws sts get-caller-identity --query Account --output text)
+
+LOGS_SCRIPT_FILE="/tmp/iam-serviceaccount.log"
```

Comment:
Could be weird with multiple users.  Maybe do something like:
```
LOGS_SCRIPT_FILE="$(mktemp "$XDG_RUNTIME_DIR/iam-serviceaccount.log_XXXXXX")"
```

Reply:
you can also use `gen3_log_info` and `gen3_log_err` to log messages to STDERR rather than send them to a log file - ex:
```
gen3_log_info "done doing whatever"
...
gen3_log_err "whatever failed"
...
```

Reply:
OK, sound like a good idea.